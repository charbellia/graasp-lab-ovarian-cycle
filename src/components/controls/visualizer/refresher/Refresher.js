import React from 'react';
import Qs from 'qs';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Setting from './Setting';

const Refresher = ({
  reloadPage,
  t,
  onOpenModal,
  openModal,
  onCloseModal,
  handleChangeComplete,
  handleLang,
  themeColor,
  toggleTitle,
}) => {
  const {
    mode = 'default',
  } = Qs.parse(window.location.search, { ignoreQueryPrefix: true });
  return (
    <div className="modal-container">
      { mode === 'default' ? (
        <Setting
          onOpenModal={onOpenModal}
          onCloseModal={onCloseModal}
          openModal={openModal}
          handleChangeComplete={handleChangeComplete}
          handleLang={handleLang}
          themeColor={themeColor}
          toggleTitle={toggleTitle}
          reloadPage={reloadPage}
          t={t}
        />
      ) : ''
      }
    </div>
  );
};


Refresher.propTypes = {
  reloadPage: PropTypes.func.isRequired,
  onOpenModal: PropTypes.func.isRequired,
  onCloseModal: PropTypes.func.isRequired,
  handleChangeComplete: PropTypes.func.isRequired,
  handleLang: PropTypes.func.isRequired,
  t: PropTypes.func.isRequired,
  openModal: PropTypes.bool.isRequired,
  themeColor: PropTypes.string.isRequired,
  toggleTitle: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  themeColor: state.setting.themeColor,
});

export default connect(mapStateToProps)(Refresher);
