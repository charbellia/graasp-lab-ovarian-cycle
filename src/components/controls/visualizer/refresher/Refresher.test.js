import React from 'react';
import { shallow } from 'enzyme';
import { Button } from 'reactstrap';
import Refresher from './Refresher';

describe('<Refresher /> component', () => {
  const props = {
    reloadPage: jest.fn(),
  };

  const component = shallow(<Refresher {...props} />);

  it('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });
  it('shows three <Button /> components', () => {
    expect(component.find(Button).length).toEqual(1);
  });
  it('has one <svg /> with class fa-redo-alt', () => {
    expect(component.find('svg.fa-redo-alt').length).toEqual(1);
  });
});
