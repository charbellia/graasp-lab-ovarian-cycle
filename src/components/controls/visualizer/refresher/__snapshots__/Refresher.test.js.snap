// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Refresher /> component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Refresher
    classes={Object {}}
    reloadPage={[MockFunction]}
    themeColor="#5050d2"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "aria-label": "Add",
      "children": <pure(RefreshIcon)
        style={
          Object {
            "color": "white",
          }
        }
      />,
      "className": undefined,
      "color": "primary",
      "onClick": [Function],
      "style": Object {
        "backgroundColor": "#5050d2",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "style": Object {
          "color": "white",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "aria-label": "Add",
        "children": <pure(RefreshIcon)
          style={
            Object {
              "color": "white",
            }
          }
        />,
        "className": undefined,
        "color": "primary",
        "onClick": [Function],
        "style": Object {
          "backgroundColor": "#5050d2",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "style": Object {
            "color": "white",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
